pkg/splunk/splunk.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		log.Entry().Warnf("Could not read branch file. %v", err)
+		contentGitHubOwner = []byte("N/A")
+	}
+	gitOwner := string(contentGitHubOwner)
+
+	return commitHash, branch, gitOwner, gitRepository
+}
+
+// MonitoringData definition for monitoring
+type MonitoringData struct {
+	PipelineUrlHash string `json:"PipelineUrlHash,omitempty"`
+	BuildUrlHash    string `json:"BuildUrlHash,omitempty"`
+	StageName       string `json:"StageName,omitempty"`
+	StepName        string `json:"StepName,omitempty"`
+	ExitCode        string `json:"ExitCode,omitempty"`
+	Duration        string `json:"Duration,omitempty"`
+	ErrorCode       string `json:"ErrorCode,omitempty"`
+	ErrorCategory   string `json:"ErrorCategory,omitempty"`
+	CorrelationID   string `json:"CorrelationID,omitempty"`
+	CommitHash      string `json:"CommitHash,omitempty"`
+	Branch          string `json:"Branch,omitempty"`
+	GitOwner        string `json:"GitOwner,omitempty"`
+	GitRepository   string `json:"GitRepository,omitempty"`
+}
+
+func prepareTelemetry(customTelemetryData telemetry.CustomData) MonitoringData {
+	tData := telemetry.GetData(&customTelemetryData)
+
+	commitHash, branch, gitOwner, gitRepository := readPipelineEnvironment()
+
+	return MonitoringData{
+		PipelineUrlHash: tData.PipelineURLHash,
+		BuildUrlHash:    tData.BuildURLHash,
+		StageName:       tData.StageName,
+		StepName:        tData.BaseData.StepName,
+		ExitCode:        tData.CustomData.ErrorCode,
+		Duration:        tData.CustomData.Duration,
+		ErrorCode:       tData.CustomData.ErrorCode,
+		ErrorCategory:   tData.CustomData.ErrorCategory,
+		CorrelationID:   SplunkClient.correlationID,
+		CommitHash:      commitHash,
+		Branch:          branch,
+		GitOwner:        gitOwner,
+		GitRepository:   gitRepository,
+	}
+}
+
+type Event struct {
+	Messages  []log.Message  `json:"messages,omitempty"`  // messages
+	Telemetry MonitoringData `json:"telemetry,omitempty"` // telemetryData
+}
+type Details struct {
+	Host       string `json:"host"`                 // hostname
+	Source     string `json:"source,omitempty"`     // optional description of the source of the event; typically the app's name
+	SourceType string `json:"sourcetype,omitempty"` // optional name of a Splunk parsing configuration; this is usually inferred by Splunk
+	Index      string `json:"index,omitempty"`      // optional name of the Splunk index to store the event in; not required if the token has a default index set in Splunk
+	Event      Event  `json:"event,omitempty"`      // throw any useful key/val pairs here}
+}
+
+func tryPostMessages(telemetryData MonitoringData, messages []log.Message) error {
+
+	event := Event{
+		Messages:  messages,
+		Telemetry: telemetryData,
+	}
+	details := Details{
+		Host:       SplunkClient.correlationID,
+		SourceType: "_json",
+		Index:      SplunkClient.splunkIndex,
+		Event:      event,
+	}
+
+	payload, err := json.Marshal(details)
+	if err != nil {
+		return err
```

Comment:
enrich `err` with more details

Reply:
Please see this: https://github.com/SAP/jenkins-library/blob/master/DEVELOPMENT.md#error-handling

Reply:
Strange. I changed this already over the weekend and In the original repo the changes are also applied:
https://github.com/continuous-monitoring/jenkins-library/blob/af77b7b20257ddb5247b0dcb8ca4e9b77e59d8a7/pkg/splunk/splunk.go#L169

Same applies for all the other error handlings which you mentioned.